archive/logic_puzzles/lady_or_tiger.lean

```diff
@@ -0,0 +1,205 @@
+/-
+Copyright (c) 2020 Dan Stanescu.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: D.Stanescu and Y. G. Kudryashov.
+-/
+
+import tactic
+
+/-!
+# Six logic puzzles.
+-/
+
+/-- The first six puzzles from:
+    "The Lady or the Tiger? And Other Logic Puzzles"
+        by Raymond Smullyan.
+First three contributed to the Lean Zulip chat by Yury G. Kudryashov
+but apparently set up by his seven-years-old son.
+Slightly modified in appearance (for readability) but not in content by D. Stanescu.
+-/
+
+inductive door_leads_to
+| lady
+| tiger
+
+notation `y` := door_leads_to.lady
+notation `n` := door_leads_to.tiger
+
+structure Q := (d₁ d₂ : door_leads_to)
+
+def Q1 := Q
```

Comment:
Presumably we need informal statements of these puzzles, too?

Reply:
They will be further down the file. Nevertheless, that's that much one can do because of possible copyright issues. I will add links to puzzle descriptions whenever available.

Reply:
Actually they were indeed missing. Added descriptions now. Please review changes.